app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -639,9 +639,15 @@ public boolean onOptionsItemSelected(MenuItem item) {
      */
     private void done(boolean failed) {
         if (Intent.ACTION_VIEW.equals(CommCareSetupActivity.this.getIntent().getAction())) {
-            //Call out to CommCare Home
-            Intent i = new Intent(getApplicationContext(), DispatchActivity.class);
-            startActivity(i);
+            if (getIntent().getBooleanExtra(FORCE_VALIDATE_KEY, false)) {

```

Comment:
I don't understand what's going on here. Why isn't the usual mm verification path sufficient here? And why is it saying that this came from the app manager when it didn't?